idx/memory/memory.go

```diff
@@ -130,6 +147,88 @@ func (m *MemoryIdx) Update(entry idx.Archive) {
 	m.Unlock()
 }
 
+// indexTags reads the tags of a given metric definition and creates the
+// corresponding tag index entries to refer to it. It assumes a lock is
+// already held.
+func (m *MemoryIdx) indexTags(def *schema.MetricDefinition) {
+	tags, ok := m.Tags[def.OrgId]
+	if !ok {
+		tags = make(TagIndex)
+		m.Tags[def.OrgId] = tags
+	}
+	for _, tag := range def.Tags {
+		tagSplits := strings.SplitN(tag, "=", 2)
+		if len(tagSplits) < 2 {
+			// should never happen because every tag in the index
+			// must have a valid format
+			InvalidTagInIndex.Inc()
+			log.Error(3, "memory-idx: Tag %s has an invalid format", tag)
```

Comment:
for all these corruption/invalid cases, we should log the metricdef id, so we can always look up the def from the index to diagnose what exactly is wrong. also i would recommend to use `%q` (quoted strings) as some chars can get confusing when printed directly

Reply:
done https://github.com/grafana/metrictank/pull/729/commits/0a887f8f61ac1f7a1a01a09331891aa4c8416994